zokrates_core/src/proof_system/bn128/utils/bellman.rs

```diff
@@ -199,6 +201,95 @@ impl Circuit<Bn256> for Computation<FieldPrime> {
     }
 }
 
+mod parse {
+    use lazy_static::lazy_static;
+
+    use super::*;
+    use regex::Regex;
+
+    lazy_static! {
+        static ref G2_REGEX: Regex = Regex::new(r"G2\(x=Fq2\(Fq\((?P<x0>0[xX][0-9a-fA-F]{64})\) \+ Fq\((?P<x1>0[xX][0-9a-fA-F]{64})\) \* u\), y=Fq2\(Fq\((?P<y0>0[xX][0-9a-fA-F]{64})\) \+ Fq\((?P<y1>0[xX][0-9a-fA-F]{64})\) \* u\)\)").unwrap();
```

Comment:
Is the check for capital `X` necessary?
According to the implementation the string representation should always start with `0x` (https://github.com/zkcrypto/ff/blob/master/ff_derive/src/lib.rs#L148)

Reply:
Good spot, I took a generic hex regex, don't think it's bad here